src/main/scala/com.snowplowanalytics/weather/providers/openweather/Client.scala

```diff
@@ -28,25 +29,23 @@ import Responses._
 import Requests._
 
 /**
- * Base client trait with defined client methods such a `historyById`, `historyByName`
+ * Base client trait with defined client methods such as `historyById`, `historyByName`
  * common for subclasses
  *
- * @tparam Response response wrapper for `Client` subclass, such as `Future`
+ * @tparam F response wrapper for `Client` subclass, such as cats `IO`
  *                  all `receive` logic should be wrapped in it
  */
-trait Client[Response[_]] {
-
-  private implicit val formats = DefaultFormats
+trait Client[F[_]] {
 
   /**
    * Main client logic for Request => Response function,
-   * where Response is wrappeed in tparam `Response`
+   * where Response is wrappeed in tparam `F`
    *
    * @param owmRequest request built by client method
    * @tparam W type of weather response to extract
-   * @return extracted either error or weather
+   * @return extracted either error or weather wrapped in `F`
    */
-  def receive[W <: OwmResponse: Manifest](owmRequest: OwmRequest): Response[W]
+  def receive[W <: OwmResponse: Decoder](owmRequest: OwmRequest): F[Either[WeatherError, W]]
```

Comment:
Decided to make `Either` explicit in all these functions, as I would argue that:

1. It's clearer to see what the function will actually return
2. It makes sense as a part of the method contract for subclasses

Let me know if you disagree

Reply:
agreed :+1: